pkg/fakesocket/fake.go

```diff
@@ -19,40 +18,73 @@ var FakeAddr = &net.UDPAddr{
 	Port: 8181,
 }
 
+var ErrClosedConnection = errors.New("Connection is closed")
+var ErrAlreadyClosedConnection = errors.New("Connection is already closed")
+
 // FakePacketConn is a fake net.PacketConn providing FakeMetric when read from.
-type FakePacketConn struct{}
+type FakePacketConn struct {
+	closed chan int
```

Comment:
Make it a `chan struct{}` to highlight that nothing valuable can be read from it. It is a convention.